### ISSUE TITLE
Include Windows Terminal in Windows Sandbox

### ISSUE THREAD

**Issue by Stanzilla** [microsoft/terminal#13870]
# Description of the new feature/enhancement

Currently, Windows Sandbox does not come with Windows Terminal, it would be nice to have it installed in there by default.

# Proposed technical implementation details (optional)

**Comment by zadjii-msft** [comment]
Huh. I kinda just imagined that Windows Sandbox would use the same build as the host OS, so if you're on Windows 11, the Terminal would be preinstalled there just the same as it is on the host...

**Comment by Stanzilla** [comment]
> Huh. I kinda just imagined that Windows Sandbox would use the same build as the host OS, so if you're on Windows 11, the Terminal would be preinstalled there just the same as it is on the host...

I thought the same but it looks like it is super stripped down, can't even install store apps etc.

**Comment by roysubs** [comment]
Windows Sandbox is a completely stripped down image. Hence, Microsoft won't include requests for application X, Y, Z to be included. The problem here is that no Store (Metro) app can be installed in Windows Sandbox (including Terminal). Hopefully, Microsoft will include Store support in a future update as this is quite a big omission, considering that it is Microsoft's own app deployment technology for Windows(!)

**Comment by carlos-zamora** [comment]
Tracking internally with MSFT:36913286

**Comment by arc-hbas** [comment]
@carlos-zamora any updates from that internal track?

**Comment by carlos-zamora** [comment]
Nope. No updates still.

**Comment by arc-hbas** [comment]
Thanks for the quick reply @carlos-zamora!

For those of you still wanting some way to execute Windows Terminal within a sandbox, I would suggest downloading and extracting the [portable ZIP](https://aka.ms/terminal-canary-zip-x64) (see details [here](https://github.com/microsoft/terminal?tab=readme-ov-file#installing-windows-terminal-canary)). You can automate this somewhat with windows sandbox by creating a `wsb` configuration file (see [here](https://learn.microsoft.com/en-us/windows/security/application-security/application-isolation/windows-sandbox/windows-sandbox-configure-using-wsb-file)) and executing a script on logon to download and extract the Windows Terminal portable zip.

#### DownloadWindowsTerminal.ps1
```ps1
Invoke-WebRequest "https://aka.ms/terminal-canary-zip-x64" -OutFile "$home\Downloads\Terminal.zip"
Expand-Archive "$home\Downloads\Terminal.zip" -DestinationPath "$env:ProgramFiles\Windows Terminal"

# Other logic to add start menu and desktop shortcuts....
```
#### Windows Sandbox Dev.wsb
```xml
<Configuration>
  <MappedFolders>
    <MappedFolder>
      <HostFolder>C:\Users\{USER}\Desktop\ForSandbox</HostFolder>
      <ReadOnly>true</ReadOnly>
    </MappedFolder>
  </MappedFolders>
  <LogonCommand>
    <!-- Logon PS script how-to: https://stackoverflow.com/a/64812427/10388359 -->
    <Command>powershell -ExecutionPolicy Unrestricted -Command "start powershell {-noexit -File C:\Users\WDAGUtilityAccount\Desktop\ForSandbox\DownloadWindowsTerminal.ps1}"</Command>
  </LogonCommand>
</Configuration>
```

Here's a [blog post](https://powershellisfun.com/2024/08/02/creating-a-development-windows-sandbox-using-powershell-and-winget/) that explains quite a bit more on how to pre-setup a dev-like environment. Author provides other scripts that generally work out of the box (minus one issue with winget).

Anyhow, thought I'd share.

Still would be nice if this worked out-of-the-box and would save us a bunch of prep time (installing winget + terminal takes ~5 mins!).

**Comment by AltitudeApps** [comment]
Is there any way to successfully install the .msixbundle while inside Windows Sandbox?

I used the command:
`dism /Online /Add-ProvisionedAppxPackage /PackagePath:"Microsoft.WindowsTerminal_1.24.10621.0_8wekyb3d8bbwe.msixbundle" /SkipLicense`

and it appeared to run successfully.  The start menu item is present, but clicking it does nothing
<br/>
<br/>
<img width="358" height="271" alt="Image" src="https://github.com/user-attachments/assets/ab1049d7-edf3-453d-8713-df9158faba17" />

<br/>
<br/>
The app execution alias is listed in Settings...
<br/>
<br/>
<img width="720" height="536" alt="Image" src="https://github.com/user-attachments/assets/5e3549b5-07b0-48bc-87a3-1bbf0075fe25" />
<br/>
<br/>
but `wt.exe` does not seem to be where I expect it might be.
<br/>
<br/>
<img width="741" height="389" alt="Image" src="https://github.com/user-attachments/assets/348a4279-03fd-4738-a8a0-705d36dbc4df" />
<br/>
<br/>
It is present on the filesystem:
<br/>
<br/>
<img width="872" height="276" alt="Image" src="https://github.com/user-attachments/assets/1762d630-2759-404f-9977-b95ba929928c" />
<br/>
<br/>
but I do not have permissions to access it.

I am able to use Windows Terminal from a folder created from the .zip archive distribution.  So that part is OK.

But I don't understand why the .msixbundle kind-of sort-of installs, but fails to run.

What do?


<hr>

update:  I installed XAML, and now it does try to run, but then this happens:

<img width="645" height="309" alt="Image" src="https://github.com/user-attachments/assets/1d9a79eb-edde-401a-aa85-568c8f133e74" />

**Comment by DHowett** [comment]
Honestly, I am pretty unclear on what happens when you add a provisioned application package in online mode. My only exposure to it is its use in audit mode or on an offline image, both of which allow for user profile load-time postinstallation...

Can you try with `Add-AppxPackage` instead? First XAML, then Terminal.

**Comment by AltitudeApps** [comment]
Yes, I just did that.   By the way, you can add a provisioned package in online mode just fine, because just before this step I installed WinGet by doing this.

So then I used WinGet to install XAML, then I installed Windows Terminal using WinGet.

No joy.  Similar results.

<img width="1061" height="731" alt="Image" src="https://github.com/user-attachments/assets/1c1a3bbf-2d68-4044-9ffe-b662f310a57e" />

<hr>

If there's no way to do this, then there's no way to do this.  That's OK.

But being able to install Windows Terminal from the .msixbundle would satisfy the original ask of this issue, in my view.

**Comment by DHowett** [comment]
In truth, I'm not sure what's going wrong in your case! It is supposed to work. I tried a straight-up provisioned package (thanks for letting me know it should work online), with results below.

<img width="2030" height="1271" alt="Image" src="https://github.com/user-attachments/assets/902e008e-0088-488e-842f-dad1687b88f7" />

**Comment by AltitudeApps** [comment]
Thanks for the response, much appreciated.

I used a script I found online to install the Store itself into the sandbox, and then Terminal started working.

https://github.com/ThioJoe/Windows-Sandbox-Tools/blob/main/Installer%20Scripts/Install-Microsoft-Store.ps1

I'm not sure what's going on either; I have certain speculations that the user permissions aren't being set up properly.  (Who knows?)   But since installing the Store results in being able to install and run Terminal, I'm guessing that one of the dependencies which is required to be present before the Store's .msixbundle is installed includes the magic sauce that it required.

I'm modifying that Install-Microsoft-Store script such that I can run it outside of the Sandbox environment and obtain all the files I need prior to starting the sandbox.  It seems silly to re-download all the files every single time an instance of the sandbox is started.

I will update this thread with the results when I'm done, provided I actually succeed and don't get distracted, lol.

I will also attempt to try installing Terminal using the method you just demonstrated, but on another Windows 11 machine.

P.S.   I will also try the exact same commands that you show above, and see what happens.

**Comment by AltitudeApps** [comment]
Some success:

<img width="1314" height="742" alt="Image" src="https://github.com/user-attachments/assets/404042e7-f9c9-4814-aedd-d39f7ef33828" />

By using that "Install-Microsoft-Store" script, using the flag `-noinstall`, I was able to download all the .appx files needed to install the store.

Then I used dism to add these appx packages to the online image, as we've been discussing.   (I did not bother to install the store itself, only the required dependencies.)

Once this was done, I was able to run Terminal, as seen in the screenshot.
